File: server/src/entertainment/entertainment.controller.ts
// src/entertainment/entertainment.controller.ts

import { Controller, Get, Param, ParseIntPipe, Query } from '@nestjs/common';
import { EntertainmentService } from './entertainment.service';

@Controller('entertainment-image')
export class EntertainmentController {
  constructor(private readonly entertainmentService: EntertainmentService) {}

  @Get(':marketId/:placeName/:isIndoor')
  async getImage(
    @Param('marketId', ParseIntPipe) marketId: number,
    @Param('placeName') placeName: string,
    @Param('isIndoor') isIndoor: string,
  ) {
    const imageUrl = await this.entertainmentService.findImageUrl(
      marketId,
      decodeURIComponent(placeName),
      isIndoor === 'true',
    );

    if (!imageUrl) {
      return { image_url: null };
    }

    return { image_url: imageUrl };
  }
}

File: backend/mergewhat/src/main/java/com/s3/mergewhat/market/controller/EntertainmentController.java
package com.s3.mergewhat.market.controller;

import com.s3.mergewhat.market.domain.aggregate.entity.Entertainment;
import com.s3.mergewhat.market.service.ImageCrawlService;
import com.s3.mergewhat.market.vo.ResponseEntertainmentVO;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/entertainment-image")
public class EntertainmentController {
    private final ImageCrawlService service;

    @GetMapping("{marketId}/{placeName}/{isIndoor}")
    public ResponseEntity<ResponseEntertainmentVO> getImage(
            @PathVariable Long marketId,
            @PathVariable String placeName,
            @PathVariable Boolean isIndoor
    ) {
        System.out.println("🎯 [요청 확인] marketId=" + marketId + ", placeName=" + placeName + ", isIndoor=" + isIndoor);

        String imageUrl = service.getOrCrawlImage(marketId, placeName, isIndoor);

        if (imageUrl == null) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
        }

        return ResponseEntity.ok(new ResponseEntertainmentVO(placeName, imageUrl));
    }
}
